dev/udf_proxy.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <sqlite3.h>
+#include <new>  //  std::launder
+#include <string>  //  std::string
+#include <functional>  //  std::function
+#include <utility>  //  std::move
+
+#include "error_code.h"
+
+namespace sqlite_orm {
+    namespace internal {
+        /*
+         *  Stores type-erased information in relation to a user-defined scalar or aggregate function object:
+         *  - name and argument count
+         *  - function pointers for construction/destruction
+         *  - function dispatch
+         *  - allocated memory and location
+         *  
+         *  As such, it also serves as a context for aggregation operations instead of using `sqlite3_aggregate_context()`.
+         */
+        struct udf_proxy {
+            using func_call_fn_t = void (*)(void* udfHandle,
+                                            sqlite3_context* context,
+                                            int argsCount,
+                                            sqlite3_value** values);
+            using final_call_fn_t = void (*)(void* udfHandle, sqlite3_context* context);
+
+            struct destruct_only_deleter {
+                template<class F>
+                void operator()(F* f) const noexcept {
+                    f->~F();
+                }
+            };
+
+            std::string name;
+            int argumentsCount;

```

Comment:
better default values be added like this `int argumentsCount = 0;`, `bool constructed = false;` and `void* const udfHandle = nullptr;` to be sure we never face trash values there

Reply:
The main reason in my head was not to duplicate the constructor for not fully C++14 compliant compilers (or certain versions thereof).